### DISCUSSION TITLE
Diffing tab behaviour

### DISCUSSION THREAD

**Discussion by NastyFlytrap** [WerWolv/ImHex#1708] [Feedback]
When having the hex editor and diffing be tabs in the same 'block', saving it as a layout and opening imhex, will default to diffing even if its the second tab and not the first. It'd be neat if imhex defaulted to the first tab of every 'block' unless the layout was saved with diffing being the selected tab. The other tabs dont seem to behave this way. 

Edit: Same goes for selecting it as a layout. Diffing is always selected as the primary tab even though its the second from the left, and it wasnt the one selected when the layout was saved.

**Comment by WerWolv** [comment]
Hey! I actually never noticed this behaviour. Other views do behave the same way though if e.g the diff view is not open.

I believe this is a ImGui thing simply. From the looks of it, the order in which the views are registered to ImHex (which also determines the order they are drawn in and also the order they open in when selecting a file) matters. It's always the view that gets drawn last that gets focus.

I honestly don't think there's much I can do about this except saving the currently focused view to the workspace and applying that on restore. However this will not really fix the issue in many cases still, only in a few.
